src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Domain/Payments/Payment.cs

```diff
@@ -7,15 +7,47 @@ public class Payment: IAggregateRoot<Guid>, IOptimisticConcurrent
     public Guid Id { get; private init; }
     public PaymentStatus PaymentStatus { get; private set; }
     public DateTime DateCreated { get; private init; }
+    public DateTime DateToExpire { get; private init; }
+    private const int ExpirationTimeInMinutes = 30;
     public DateTime DateModified { get; set; }
 
     public Payment()
     {
         Id = Guid.NewGuid();
         PaymentStatus = PaymentStatus.Created;
         DateCreated = DateTime.UtcNow;
+        DateToExpire = DateCreated.AddMinutes(ExpirationTimeInMinutes);
     }
 
+    public virtual bool ConfirmPayment()
+    {
+        if (PaymentAlreadyDecidedOrExpired())
+        {
+            return false;
+        }
+        PaymentStatus = PaymentStatus.Confirmed;
+        return true;
+    }
+    
+    public virtual bool CancelPayment()
+    {
+        if (PaymentAlreadyDecidedOrExpired())
+        {
+            return false;
+        }
+        PaymentStatus = PaymentStatus.Cancelled;
+        return true;
+    }
+
+    private bool PaymentAlreadyDecidedOrExpired()
+    {
+        var hasExpired = DateTime.UtcNow > DateToExpire;
+        if (PaymentStatus is PaymentStatus.Created && hasExpired)
+        {
+            PaymentStatus = PaymentStatus.Expired;
```

Comment:
Don't change values in method that is meant to check something.

Reply:
Could you please be more clear what the issue is? Is it the name of the function? It's main purpose was to extract duplicated logic from CancelPayment and ConfirmPayment really. To alleviate the issue I can extract the update into `UpdatePaymentExpiration`  method and rename the method to `IfPaymentAlreadyDecidedOrExpired`. If the problem is with this logic being in those two functions I can put it in both property accessors.

Reply:
Addressed in 7f75126c.